File: app/src/main/java/com/example/capture/Foreground.java
package com.example.capture;

import android.app.Activity;
import android.app.Application;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public class Foreground implements Application.ActivityLifecycleCallbacks {
    private static Foreground instance;
    public static void init(Application app) {
        if (instance == null) {
            instance = new Foreground();
            app.registerActivityLifecycleCallbacks(instance);
        }
    }

    public static Foreground get() {
        return instance;
    }

    public Foreground() {
    }

    private static AppStatus mAppStatus;

    public AppStatus getAppStatus() {
        return mAppStatus;
    }

    // check if app is return foreground
     public static boolean isBackground() {
        return mAppStatus.ordinal() == AppStatus.BACKGROUND.ordinal();
    }

    public enum AppStatus {
        BACKGROUND, // app is background
        RETURNED_TO_FOREGROUND, // app returned to foreground(or first launch)
        FOREGROUND; // app is foreground
    }

    // running activity count
    private int running = 0;

    @Override
    public void onActivityCreated(@NonNull Activity activity, @Nullable Bundle savedInstanceState) {
    }

    @Override
    public void onActivityStarted(@NonNull Activity activity) {
        if (++running == 1) {
            mAppStatus = AppStatus.RETURNED_TO_FOREGROUND;        }
        else if (running > 1) {
            mAppStatus = AppStatus.FOREGROUND;
        }
    }

    @Override
    public void onActivityResumed(@NonNull Activity activity) {

    }

    @Override
    public void onActivityPaused(@NonNull Activity activity) {

    }

    @Override
    public void onActivityStopped(@NonNull Activity activity) {
        if (--running == 0) {
            mAppStatus = AppStatus.BACKGROUND;
        }
    }

    @Override
    public void onActivitySaveInstanceState(@NonNull Activity activity, @NonNull Bundle outState) {

    }

    @Override
    public void onActivityDestroyed(@NonNull Activity activity) {

    }
}

File: app/src/main/java/com/example/mediaplayer/Foreground.kt
package com.example.mediaplayer

import android.app.Activity
import android.app.Application
import android.app.Application.ActivityLifecycleCallbacks
import android.os.Bundle

class Foreground : ActivityLifecycleCallbacks {
    fun getAppStatus(): AppStatus? {
        return mAppStatus
    }

    enum class AppStatus {
        BACKGROUND,  // app is background
        RETURNED_TO_FOREGROUND,  // app returned to foreground(or first launch)
        FOREGROUND // app is foreground
    }

    // running activity count
    private var running: Int = 0
    override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle?) {}
    override fun onActivityStarted(activity: Activity) {
        if (++running == 1) {
            mAppStatus = AppStatus.RETURNED_TO_FOREGROUND
        } else if (running > 1) {
            mAppStatus = AppStatus.FOREGROUND
        }
    }

    override fun onActivityResumed(activity: Activity) {}
    override fun onActivityPaused(activity: Activity) {}
    override fun onActivityStopped(activity: Activity) {
        if (--running == 0) {
            mAppStatus = AppStatus.BACKGROUND
        }
    }

    override fun onActivitySaveInstanceState(activity: Activity, outState: Bundle) {}
    override fun onActivityDestroyed(activity: Activity) {}

    companion object {
        private var instance: Foreground? = null
        fun init(app: Application) {
            if (instance == null) {
                instance = Foreground()
                app.registerActivityLifecycleCallbacks(instance)
            }
        }

        fun get(): Foreground? {
            return instance
        }

        private var mAppStatus: AppStatus? = null

        // check if app is return foreground
        fun isBackground(): Boolean {
            return mAppStatus!!.ordinal == AppStatus.BACKGROUND.ordinal
        }
    }
}